### ISSUE TITLE
reply_to_last from Residently now returns error after being merged into master

### ISSUE THREAD

**Issue by Chadh13** [intercom/intercom-ruby#445]
The `reply_to_last` functionality seems to have stopped working after the recent merge into master and release. 

I was previously using the forked branch from Residently who implemented this feature. This feature and functionality worked before the forked branch was merged. 

After upgrading my intercom-ruby client from their no longer available fork, calling the method has now begun throwing an error in my app.

I've downgraded to Residently's fork with Intercom 3.6.2 and the `reply_to_last` feature in my app is now working again.

## Version info
  - intercom-ruby version: intercom (3.7.0)
  - Ruby version: ruby 2.5.3p105

## Expected behavior

My expectation is that when I call `reply_to_last` on a collection of conversations then the client would post a comment to the last conversation of a particular user.

```
@intercom.conversations.reply_to_last(message_type: 'comment', user_id: '00000', type: 'user', body: "Hi ")
````

## Actual behavior

```
NoMethodError: undefined method `reply_to_last' for #<Intercom::Service::Conversation:0x0000558f47398970>
```

## Steps to reproduce
 1. Create a user in intercom
 2. Create a conversation for that user
 3. Initialize the intercom client
 4. Attempt to call the `reply_to_last` method on the user's conversations

## Logs
```
[189] pry(main)>@intercom ||= Intercom::Client.new(token: key)
[190] pry(main)> @intercom.conversations.reply_to_last(message_type: 'comment', user_id: '00000', type: 'user', body: "Hi ")
NoMethodError: undefined method `reply_to_last' for #<Intercom::Service::Conversation:0x0000558f4f6f90a8>
from (pry):187:in `<main>
```

**Comment by jonnyom** [comment]
Hi @Chadh13 thanks for reporting this!

Unfortunately I'm not able to reproduce this at all at the moment unfortunately

```shell
~/s/intercom-ruby ❯❯❯ gem install intercom                                                                     master
Fetching: intercom-3.7.0.gem (100%)
Successfully installed intercom-3.7.0
Parsing documentation for intercom-3.7.0
Installing ri documentation for intercom-3.7.0
Done installing documentation for intercom after 1 seconds
1 gem installed
```

```ruby
[3] Intercom(GEM)> client = Intercom::Client.new(token: token)
[4] Intercom(GEM)> client.conversations.reply_to_last(message_type: "comment", user_id: "proper_test", type: "user", body: "TEST FROM THE API")
=> #<Intercom::Conversation:0x00007facb52397e0
 @assignee=#<Intercom::NobodyAdmin:0x00007facb5220b78 @changed_fields=#<Set: {}>, @id=nil, @type="nobody_admin">,
 @changed_fields=#<Set: {}>,
 @conversation_message=
  #<Intercom::ConversationMessage:0x00007facb5231270
   @attachments=[],
   @author=
    #<Intercom::User:0x00007facb522a8f8
     @changed_fields=#<Set: {}>,
     @custom_attributes={},
     @id="5963608285bc3bc4f5b2148b",
     @type="user">,
   @body="<p>oi oi</p>",
   @changed_fields=#<Set: {}>,
   @id="134046234",
   @subject="",
   @type="conversation_message",
   @url=nil>,
 @conversation_parts=
  [#<Intercom::ConversationPart:0x00007facb52192d8
    @assigned_to=nil,
    @attachments=[],
    @author=#<Intercom::Admin:0x00007facb520be30 @changed_fields=#<Set: {}>, @id="1309092", @type="admin">,
    @body=nil,
    @changed_fields=#<Set: {}>,
    @created_at=1541777153,
    @external_id=nil,
    @id="2282819083",
    @notified_at=1541777153,
    @part_type="close",
    @type="conversation_part",
    @updated_at=1541777153>,
   #<Intercom::ConversationPart:0x00007facb5209ba8
    @assigned_to=nil,
    @attachments=[],
    @author=
     #<Intercom::User:0x00007facb5209220
      @changed_fields=#<Set: {}>,
      @custom_attributes={},
      @id="5963608285bc3bc4f5b2148b",
      @type="user">,
    @body="<p>Hi</p>",
    @changed_fields=#<Set: {}>,
    @created_at=1542730282,
    @external_id=nil,
    @id="2322852368",
    @notified_at=1542730282,
    @part_type="comment",
    @type="conversation_part",
    @updated_at=1542730282>,
   #<Intercom::ConversationPart:0x00007facb5208500
    @assigned_to=nil,
    @attachments=[],
    @author=
     #<Intercom::User:0x00007facb5203b18
      @changed_fields=#<Set: {}>,
      @custom_attributes={},
      @id="5963608285bc3bc4f5b2148b",
      @type="user">,
    @body="<p>TEST FROM THE API</p>",
    @changed_fields=#<Set: {}>,
    @created_at=1542730316,
    @external_id=nil,
    @id="2322855517",
    @notified_at=1542730316,
    @part_type="comment",
    @type="conversation_part",
    @updated_at=1542730316>,
   #<Intercom::ConversationPart:0x00007facb5202df8
    @assigned_to=nil,
    @attachments=[],
    @author=
     #<Intercom::User:0x00007facb5202470
      @changed_fields=#<Set: {}>,
      @custom_attributes={},
      @id="5963608285bc3bc4f5b2148b",
      @type="user">,
    @body="<p>TEST FROM THE API</p>",
    @changed_fields=#<Set: {}>,
    @created_at=1542730416,
    @external_id=nil,
    @id="2322864999",
    @notified_at=1542730416,
    @part_type="comment",
    @type="conversation_part",
    @updated_at=1542730416>],
 @conversation_rating=
  {"rating"=>nil,
   "remark"=>nil,
   "created_at"=>nil,
   "customer"=>{"type"=>nil, "id"=>nil},
   "teammate"=>{"type"=>nil, "id"=>nil}},
 @created_at=1506086741,
 @customers=[{"type"=>"user", "id"=>"5963608285bc3bc4f5b2148b"}],
 @id="11929500354",
 @open=true,
 @read=true,
 @snoozed_until=nil,
 @state="open",
 @tags=[],
 @type="conversation",
 @updated_at=1542730416,
 @user=
  #<Intercom::User:0x00007facb5222c48
   @changed_fields=#<Set: {}>,
   @custom_attributes={},
   @id="5963608285bc3bc4f5b2148b",
   @type="user">,
 @waiting_since=1542730282>
```

You can see for the last two conversation parts they are successfully working for this user. Can you double check to make sure you have indeed upgraded to `3.7.0` please? 

As this seems to be working on my end I'm going to close this out, but feel free to reopen if you can reliably reproduce this on the latest version! 😄

**Comment by Chadh13** [comment]
Hey @jonnyom, I'm not certain why this issue exists however upgrading the gem still results in a failure here.

One thing I did notice is that in your stack trace it shows:
```
#<Intercom::Conversation:0x00007facb52397e0
```
Where in mine it shows:
```
NoMethodError: undefined method `reply_to_last' for 
#<Intercom::Service::Conversation:0x00007fd9c13f0820>
```

Here is my test:
```
[1] pry(main)> key = Rails.env.production? ? ENV['INTERCOM_API_KEY'] : ENV['INTERCOM_API_TEST_KEY']
"SOME KEY"
[2] pry(main)> client = Intercom::Client.new(token: key)
#<Intercom::Client:0x00007ff459f6bdf0 @username_part="dG9rOmQwYmNhZGFlX2VlMGNfNDk3Zl84MWVmXzUyZDIxM2NlNDViMjoxOjA=", @password_part="", @base_url="https://api.intercom.io", @rate_limit_details={}, @handle_rate_limit=false, @timeouts={:open_timeout=>30, :read_timeout=>90}>
[3] pry(main)> client.conversations.reply_to_last(message_type: "comment", intercom_user_id: "5ba9b5f4d1e4893f5678a4ff", type: "user", body: "TEST FROM THE API")

NoMethodError: undefined method `reply_to_last' for #<Intercom::Service::Conversation:0x00007ff459f296f8>
from (pry):3:in `<main>'
```

Notes: 
- This is being used within the Rails framework.
- Bundle pristine was run during the gem's upgrade.
- This was tested both with an intercom user who had 1 existing conversation, and more than 1 existing conversation.
- This was tested both with `user_id` and `intercom_user_id` as the lookup field.

**Comment by jonnyom** [comment]
Hey @Chadh13 thanks! I've just booted this up in a separate Rails app and you're right, it looks like there's something funky happening going on specifically when this is loaded into Rails.
```ruby
[2] pry(main)> client = Intercom::Client.new(token: token)
=> #<Intercom::Client:0x00007fe8634c07b8
 @base_url="https://api.intercom.io",
 @handle_rate_limit=false,
 @password_part="",
 @rate_limit_details={},
 @timeouts={:open_timeout=>30, :read_timeout=>90},
 @username_part="[FILTERED]">
[3] pry(main)> client.conversations.reply_to_last(message_type: "comment", user_id: "proper_test", type: "user", body: "TEST FROM RAILS CONSOLE")
NoMethodError: undefined method `reply_to_last' for #<Intercom::Service::Conversation:0x00007fe860b07c78>
from (pry):3:in `__pry__'
[4] pry(main)> Intercom::VERSION
=> "3.7.0"
```

I'm going to reopen this so we can keep digging 👍

**Comment by jonnyom** [comment]
@Chadh13 do you happen to have a link to Residently's 3.6.2 fork that you mentioned? I'm not able to find that specific functionality in the fork that he has on his repo. I'd like to compare the differences between what we've added and what he has, if any.

**Comment by jonnyom** [comment]
OK thanks @Chadh13 it looks like the last release never bundled this functionality for some reason.

I'm not sure why that would be the case, but I'm going to do a new release to 3.7.1 to make this work for you 👍

**Comment by jonnyom** [comment]
@Chadh13 I've just released a new version `3.7.1` to Rubygems, which you can now download and install, with this functionality correctly operating.

Going to close this out now as we look to be good 👍
